satip/utils.py

```diff
@@ -214,19 +214,12 @@ def load_cloudmask_to_dataset(
         ]
     )
     try:
-        # Selected bounds empirically for have no NaN values from off disk image,
+        # Selected bounds empirically that have no NaN values from off disk image,
         # and are covering the UK + a bit
         dataarray: xr.DataArray = convert_scene_to_dataarray(
             scene, band="cloud_mask", area=area, calculate_osgb=calculate_osgb
         )
-
-        # Compress and return
-        dataarray = dataarray.transpose("time", "y_geostationary", "x_geostationary", "variable")
-        dataarray = dataarray.round().clip(min=0, max=3).astype(np.int8)
-        dataarray.attrs = serialize_attrs(dataarray.attrs)
-        # Convert 3's to NaNs as they should be No Data/Space
-        dataarray = dataarray.where(dataarray["variable"] != 3)
```

Comment:
I think this line implicitly converted the cloud masks back to `float64` (because `int8` can't represent NaNs)?

Reply:
Ah, possibly, the cloud masks, when I open them up, are int8, so possibly something else is happening to the NaNs